sklearn/neighbors/regression.py

```diff
@@ -17,12 +17,13 @@
 
 from .base import _get_weights, _check_weights, NeighborsBase, KNeighborsMixin
 from .base import RadiusNeighborsMixin, SupervisedFloatMixin
-from ..base import RegressorMixin
+from ..base import RegressorMixin, RequiresPositiveDataMixin
 from ..utils import check_array
 
 
 class KNeighborsRegressor(NeighborsBase, KNeighborsMixin,
                           SupervisedFloatMixin,
+                          RequiresPositiveDataMixin,
```

Comment:
Why did you add this here ? `KNeighborsRegressor` does not require positive data in the default case.

Reply:
Yes, that was not the best idea. I did it to fix [this test](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/utils/tests/test_estimator_checks.py#L481). I see now it is using `KNeighborsRegressor(metric='precomputed')` and failing, while `KNeighborsRegressor()` passes. I'll remove the metric parameter here.